09-maze_solver/builder/main.go

```diff
@@ -24,28 +23,30 @@ func main() {
 	height, _ := strconv.Atoi(os.Args[2])
 
 	maze := generateMaze(width, height)
-	saveToPNG(maze, "maze.png")
+	saveToPNG(maze, fmt.Sprintf("maze%d_%d.png", width, height))
 }
 
 func generateMaze(width int, height int) *image.RGBA {
 	img := image.NewRGBA(image.Rect(0, 0, width, height))
 
-	// I see a red door...
 	for y := 0; y < height; y++ {
 		for x := 0; x < width; x++ {
-			img.Set(x, y, wall)
+			img.Set(x, y, defaultColours().wallColour)
 		}
 	}
 
 	entry := pos{0, height / 2}
-	img.Set(entry.x, entry.y, path)
+	img.Set(entry.x, entry.y, defaultColours().pathColour)
 
 	// draw the path
 	p := posWithCount{entry, 0}
 
 	const complexity = 2 // change this for easier / creatable mazes, or harder ones.
 	// create a massive channel, because I don't want to start a listener right now.
-	b := &builder{ps: make(chan posWithCount, width*height), width: width - 1, height: height - 1, complexity: complexity * (width + height)}
+	b := &builder{
+		ps: make(chan posWithCount, width*height), width: width - 1, height: height - 1, complexity: complexity * (width + height),
+		conf: defaultColours(),
```

Comment:
If feels weird to have `defaultColours().pathColour` earlier, and `b.conf.pathColour` later. I suggest we move the outer walls + entry to after this line so we can write `img.Set(entrance.x, entrance.y, b.conf.entranceColour)`